### PR TITLE
evm: allow setting pre-defined accounts in genesis

### PR DESCRIPTION
**Pull request by sorpaas**
This adds the genesis config `accounts` that allows setting the full values of pre-defined accounts with code and storage.

### CODE REVIEW

File: frame/evm/src/lib.rs [hunk 1]
```diff
@@ -137,12 +140,43 @@ pub trait Trait: frame_system::Trait + pallet_timestamp::Trait {
 	}
 }
 
+#[derive(Clone, Eq, PartialEq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
+/// Account definition used for genesis block construction.
```
Comment (bkchr) [inline frame/evm/src/lib.rs:145]:
```suggestion
/// Account definition used for genesis block construction.
#[derive(Clone, Eq, PartialEq, codec::Encode, codec::Decode, Debug, serde::Serialize, serde::Deserialize)]
#[cfg(feature = "std")]
```

Genesis types are only required in native.

File: frame/evm/src/lib.rs [hunk 2]
```diff
@@ -24,7 +24,10 @@ mod backend;
 
 pub use crate::backend::{Account, Log, Vicinity, Backend};
 
-use sp_std::{vec::Vec, marker::PhantomData};
+use sp_std::{vec::Vec, marker::PhantomData, collections::btree_map::BTreeMap};
+#[cfg(feature = "std")]
+use serde::{Serialize, Deserialize};
```
Comment (bkchr) [inline frame/evm/src/lib.rs:29]:
```suggestion
```

File: frame/evm/src/lib.rs [hunk 3]
```diff
@@ -24,7 +24,10 @@ mod backend;
...
+use codec::{Encode, Decode};
```
Comment (bkchr) [inline frame/evm/src/lib.rs:30]:
```suggestion
```

File: frame/evm/src/lib.rs [hunk 4]
```diff
@@ -24,7 +24,10 @@ mod backend;
 
 pub use crate::backend::{Account, Log, Vicinity, Backend};
 
-use sp_std::{vec::Vec, marker::PhantomData};
+use sp_std::{vec::Vec, marker::PhantomData, collections::btree_map::BTreeMap};
```
Comment (bkchr) [inline frame/evm/src/lib.rs:27]:
```suggestion
use sp_std::{vec::Vec, marker::PhantomData};
```

File: frame/evm/src/lib.rs [hunk 5]
```diff
@@ -137,12 +140,43 @@ pub trait Trait: frame_system::Trait + pallet_timestamp::Trait {
...
+#[derive(Clone, Eq, PartialEq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
+/// Account definition used for genesis block construction.
+pub struct GenesisAccount {
+	/// Account nonce.
+	pub nonce: U256,
+	/// Account balance.
+	pub balance: U256,
+	/// Full account storage.
+	pub storage: BTreeMap<H256, H256>,
```
Comment (bkchr) [inline frame/evm/src/lib.rs:152]:
```suggestion
	pub storage: std::collections::BTreeMap<H256, H256>,
```

**Comment by bkchr** [comment]
bot merge

**Comment by sorpaas** [comment]
bot merge

**Comment by sorpaas** [comment]
bot merge

**Comment by sjeohp** [comment]
Apologies, still ironing out some kinks in the bot

**Comment by sjeohp** [comment]
bot merge

**Comment by sjeohp** [comment]
bot merge

**Pull request merged**